reflectable/lib/src/builder_implementation.dart

```diff
@@ -2234,12 +2234,25 @@ class _ReflectorDomain {
       declarationsCode = _formatAsConstList('int', declarationsIndices);
     }
 
-    // TODO(sigurdm) clarify: Find out how to get good uri's in a
-    // builder.
-    String uriCode =
-        (_capabilities._supportsUri || _capabilities._supportsLibraries)
-            ? "Uri.parse(r'reflectable://$libraryIndex/$library')"
-            : 'null';
+    // URIs may not work at run time, but we use the `assetId` to get a
+    // plausible URI when possible.
+    String uriCode;
+    if (_capabilities._supportsUri || _capabilities._supportsLibraries) {
+      AssetId? assetId;
+      try {
+        assetId = await _resolver.assetIdForElement(library);
+      } catch (_) {
+        assetId = null;
+      }
+      if (assetId != null) {
+        var uri = assetId.uri.replace(scheme: 'package');
```

Comment:
Reading https://pub.dev/documentation/build/latest/build/AssetId/uri.html I would expect this to already be a package-uri for practical purposes. And if not a package-uri I'm not sure we want to rewrite the scheme (maybe just replace by some placeholder...).

Reply:
Interesting, I should have looked that up. ;-)

I noticed that the URI obtained using `AssetId.uri` is `asset:test_reflectable/test/reflect_type_test.dart` for the test 'reflect_type_test.dart' in the package `test_reflectable`.

The 'uri.replace(scheme: 'package')' actually won't work as a package uri an Dart source code. So maybe we should simply keep the 'asset' scheme, and leave it to users of this feature to look up how to interpret an `asset` URI.

Reply:
That sounds good.

I guess there is no package-uri for anything outside lib/ so it kind-of makes sense (and makes writing the tests a bit harder).

It would be good to leave a comment about how these Uris are generated (perhaps with a pointer to https://pub.dev/documentation/build/latest/build/Resolver/assetIdForElement.html), that would make it easier to consume.

Also the documentation could mention that package-uris can be looked up via: https://api.dart.dev/stable/2.19.2/dart-isolate/Isolate/packageConfig.html .

Reply:
Or maybe better: https://api.dart.dev/stable/2.19.2/dart-isolate/Isolate/resolvePackageUri.html

Reply:
Thanks! -- added those links to CHANGELOG.md.

Reply:
I would expect it to go here: https://pub.dev/documentation/reflectable/latest/reflectable.mirrors/LibraryMirror/uri.html